### PR TITLE
[luci-interpreter] Add support for relu activation

### PR DESCRIPTION
**Pull request by ai-moiseev**
Add support for relu activation op in luci-interpreter.

ONE-DCO-1.0-Signed-off-by: Alexander Moiseev <a.moiseev@samsung.com>

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/Relu.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+void Relu::evalQuantized() const
+{
+  const auto input_data = getTensorData<uint8_t>(input());
+  const auto input_zero_point = input()->zero_point();
+  const auto size = getTensorShape(input()).FlatSize();
+  auto output_data = getTensorData<uint8_t>(output());
+
+  for (auto i = decltype(size){0}; i < size; ++i)
+  {
+    output_data[i] = input_data[i] >= input_zero_point ? input_data[i] : input_zero_point;
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/Relu.cpp:72]: This does not seem what TFLite does in U8 case.
IIUC, input and output quantization parameters do not have to match. If they are different, the value should be "requantized".
Please check out `ReluPrepare` and `ReluEval` in `lite/kernels/activations.cc`.

File: compiler/luci-interpreter/src/kernels/Relu.test.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+{
+namespace kernels
+{
+namespace
+{
+
+using namespace testing;
+
+template <typename T>
+void Check(std::initializer_list<int32_t> input_shape, std::initializer_list<int32_t> output_shape,
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/Relu.test.cpp:31]: In other tests, this function is used to check several data types (typename T). You use it only for `float`, so why don't just inline it?)

File: compiler/luci-interpreter/src/kernels/Relu.test.cpp [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+  Relu kernel(&input_tensor, &output_tensor);
+  kernel.configure();
+  kernel.execute();
+
+  EXPECT_THAT(dequantize(extractTensorData<uint8_t>(output_tensor), output_tensor.scale(),
+                         output_tensor.zero_point()),
+              ElementsAreArray(ArrayFloatNear(ref_output_data, kQuantizedTolerance)));
+  EXPECT_THAT(extractTensorShape(output_tensor), ::testing::ElementsAreArray({1, 2, 3, 1}));
+
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/Relu.test.cpp:92]: Can you add the test for the quantized value for uint8 kernels (e.g., like [logsoftmax](https://github.com/Samsung/ONE/blob/master/compiler/luci-interpreter/src/kernels/LogSoftmax.test.cpp#L81-L82))?

If it is difficult to make reference data, using the test data in [TFLite](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/kernels/activations_test.cc#L661-L682) will be ok.

File: compiler/luci-interpreter/src/kernels/Relu.test.cpp [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  SUCCEED();
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/Relu.test.cpp:93]: Seems unnecessary.

File: compiler/luci-interpreter/src/kernels/Relu.cpp (hunk 1 above)
Comment (ai-moiseev) [reply compiler/luci-interpreter/src/kernels/Relu.cpp]: Done, replaced with call to quantized ReluX method. Also added test-case for re-quantization.

File: compiler/luci-interpreter/src/kernels/Relu.test.cpp (hunk 2 above)
Comment (ai-moiseev) [reply compiler/luci-interpreter/src/kernels/Relu.test.cpp:31]: Done, inlined to test case body.

File: compiler/luci-interpreter/src/kernels/Relu.test.cpp (hunk 3 above)
Comment (ai-moiseev) [reply compiler/luci-interpreter/src/kernels/Relu.test.cpp:134]: Done, I've added two `uint8_t` value checking test cases.

File: compiler/luci-interpreter/src/kernels/Relu.test.cpp (hunk 4 above)
Comment (ai-moiseev) [reply compiler/luci-interpreter/src/kernels/Relu.test.cpp]: Done, removed it. Was taken from other test cases, thought it was some kind of convention :)

Comment (s-barannikov) [reply compiler/luci-interpreter/src/kernels/Relu.test.cpp]: It is used for static analysis somehow. If the test body does not contain any of "EXPECT_*", "ASSERT_*", then the test is considered invalid, despite of the fact that they could be called function called from the test.

File: compiler/luci-interpreter/src/kernels/Relu.test.cpp (hunk 2 above)
Comment (s-barannikov) [reply compiler/luci-interpreter/src/kernels/Relu.test.cpp:31]: It seems unused now -_-

**Review by jinevening — DISMISSED** [review]
LGTM (`Check` function in the test code needs to be removed)

**Comment by ai-moiseev** [comment]
> LGTM (`Check` function in the test code needs to be removed)

@jinevening Done, removed `Check` from `*.test.cpp`

**Review by jinevening — APPROVED** [review]

**Review by s-barannikov — APPROVED** [review]

**Pull request merged**